crates/forge_loader/src/manifest.rs

```diff
@@ -208,63 +282,102 @@ impl<T> AsRef<T> for FunctionTy<T> {
 }
 
 impl<'a> ForgeModules<'a> {
-    pub fn into_analyzable_functions(
-        mut self,
-    ) -> impl Iterator<Item = FunctionTy<FunctionMod<'a>>> {
-        //FIXME: The logic here is incorrect, a function should be in the ignored function IIF it
-        //is not invoked by a user-invocable module.
-
+    // TODO: function returns iterator where each item is some specified type.
+    pub fn into_analyzable_functions(mut self) -> impl Iterator<Item = Entrypoint<'a>> {
```

Comment:
<!--
 Copyright 2024 Joshua Wong.
 SPDX-License-Identifier: Apache-2.0
-->

There's a possibility that whenever we add new invokable modules to the `ForgeModules` struct that we forget to add them to the logic here. If instead of accessing properties using `self.<prop>`, we do:

```rust
let Self { webtriggers, custom_field, etc } = self;
```

Then `rustc` will throw a compile error if we do not explicitly destructure it like above. That way, we won't forget to add them here. If a module is irrelevant to the logic below, we can destructure the property, but not bind it like so:

```rust
let Self {
    event_triggers: _, // NOTE: non-invokable module.
} = self;
```

Make sure that you don't use a catch-all pattern like `..` at the end, as that would defeat the purpose of the explicit destructuring.